test/Optimizer/fgpeepbug.js

```diff
@@ -0,0 +1,15 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+function foo() {
+    var a = {b: {}}
+    a = null !== a.b && 0 < a.b.a
+    if (!a) return a
+}
+foo();

```

Comment:
Since you are not running this test with any flags, could you make it a loop of ~100?
Also, could you validate the return value (assuming that was the bug originally). I think it should return `false` right ?

Reply:
UTs run with `-maxInterpretCount:1 -maxSimpleJitRunCount:1`, which is why I have 3 calls

Reply:
The bug was that we load `false` into `a`, and then do `a.b.a`, which throws an exception